src/components/Menu/Menu.tsx

```diff
@@ -49,33 +55,60 @@ export type MenuItemsProps = ExtractProps<typeof HeadlessMenu.Items>;
  *
  * A dropdown that reveals or hides a list of actions
  */
-export const Menu = ({ className, ...other }: MenuProps) => {
+export const Menu = ({
+  buttonText,
+  children,
+  className,
+  ...other
+}: MenuProps) => {
   const menuClassNames = clsx(className, styles['menu']);
-  return <HeadlessMenu as="div" className={menuClassNames} {...other} />;
+  if (typeof children === 'function') {
+    return (
+      <HeadlessMenu as="div" className={menuClassNames} {...other}>
+        {({ open, close }) => (
+          <>
+            {buttonText && (
+              <MenuButton className={styles['menu__button']}>
+                {buttonText}
+                <Icon
+                  className={styles['menu__button--icon']}
+                  name="expand-more"
+                  purpose="decorative"
+                  size="1.25rem"
+                />
+              </MenuButton>
+            )}
+            {children({ open, close })}
+          </>
+        )}
+      </HeadlessMenu>
+    );
+  }
+  return (
+    <HeadlessMenu as="div" className={menuClassNames} {...other}>
+      {buttonText && (
+        <MenuButton className={styles['menu__button']}>
+          {buttonText}
+          <Icon
+            className={styles['menu__button--icon']}
+            name="expand-more"
+            purpose="decorative"
+            size="1.25rem"
+          />
+        </MenuButton>
+      )}
+      {children}
+    </HeadlessMenu>
+  );
 };
 
 /**
  * A button that when clicked, shows or hides the Options
+ * Refer to Button documentation for more style variants.
  */
-const MenuButton = ({ children, className, ...other }: MenuButtonProps) => {
-  const buttonClassNames = clsx(styles['menu__button'], className);
+const MenuButton = ({ status, ...other }: MenuButtonProps) => {
   return (
-    <HeadlessMenu.Button
-      as={ClickableStyle}
```

Comment:
Originally was `<ClickableStyle>` but changed it to `<Button>` since the alternative (link) is not an appropriate trigger for a menu

Reply:
I think this makes sense. The rationale here was that, since Headless gives you a `<button>`, this composed a styled button in the same way `Button` does internally. If we give it `Button` directly, looks like we get the same result. using `ClickableStyle` composes one less thing I guess

What things can be used as a menu trigger?

Reply:
So far we have text in the mocks, and the vertical dots icon in GST